### DISCUSSION TITLE
postgres table layout

### DISCUSSION THREAD

**Discussion by jeremy-windsor** [BerriAI/litellm#5795] [Q&A] [answered]
I'm trying to set up a simple postgres table for litellm proxy and I keep getting errors stating certain tables don't exist.

22:34:40 - LiteLLM Proxy:ERROR: proxy_server.py:2338 - litellm.proxy_server.py::add_deployment() - Error getting new models from DB - The table `public.LiteLLM_ProxyModelTable` does not exist in the current database.

22:34:40 - LiteLLM Proxy:ERROR: proxy_server.py:2189 - Error adding/deleting model to llm_router: The table `public.LiteLLM_Config` does not exist in the current database.

Is there documentation around getting this up and working with a brand new postgres database?

Thanks

**Comment by ishaan-jaff** [comment, marked as answer]
the litellm docker image auto-creates the tables for you - are you using our docker image @isriam ?

**Comment by jeremy-windsor** [comment]
the problem was my postgres permissions for the user i had for litellm, thank you!

**Comment by stdweird** [comment]
@isriam @ishaan-jaff  what postgresql permissions are required?

**Comment by jaceyang97** [comment]
@isriam 

tried
`GRANT USAGE, CREATE ON SCHEMA public TO public;`

`GRANT SELECT, INSERT, UPDATE, DELETE ON ALL TABLES IN SCHEMA public TO public;`

`ALTER DEFAULT PRIVILEGES IN SCHEMA public GRANT SELECT, INSERT, UPDATE, DELETE ON TABLES TO public;`

none of them worked.

what worked for you?
